atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/QosCassandraEteTestSetup.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.ete;
+
+
+import java.net.InetSocketAddress;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.cassandra.ImmutableCassandraCredentialsConfig;
+import com.palantir.atlasdb.cassandra.ImmutableCassandraKeyValueServiceConfig;
+import com.palantir.atlasdb.config.AtlasDbConfig;
+import com.palantir.atlasdb.config.AtlasDbRuntimeConfig;
+import com.palantir.atlasdb.config.ImmutableAtlasDbConfig;
+import com.palantir.atlasdb.config.ImmutableAtlasDbRuntimeConfig;
+import com.palantir.atlasdb.config.ImmutableSweepConfig;
+import com.palantir.atlasdb.factory.TransactionManagers;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.qos.config.ImmutableQosClientConfig;
+import com.palantir.atlasdb.qos.config.ImmutableQosLimitsConfig;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.atlasdb.todo.TodoSchema;
+import com.palantir.atlasdb.transaction.impl.SerializableTransactionManager;
+import com.palantir.atlasdb.util.AtlasDbMetrics;
+import com.palantir.docker.compose.DockerComposeRule;
+import com.palantir.docker.compose.configuration.ShutdownStrategy;
+import com.palantir.docker.compose.connection.Container;
+import com.palantir.docker.compose.connection.DockerPort;
+import com.palantir.docker.compose.logging.LogDirectory;
+import com.palantir.remoting.api.config.service.HumanReadableDuration;
+
+public class QosCassandraEteTestSetup {
+    private static final Random random = new Random();
+    static SerializableTransactionManager serializableTransactionManager;
+    static final int readBytesPerSecond = 10_000;
+    static final int writeBytesPerSecond = 10_000;
+    private static final int CASSANDRA_PORT_NUMBER = 9160;
+    static final int MAX_SOFT_LIMITING_SLEEP_MILLIS = 2000;
```

Comment:
In general I think the non-private ones should probably be protected rather than package visibility (the qos cassandra read test works because it's in the same package, but if I extend this elsewhere I probably want to be able to access some of these).

Reply:
Intellij kept warning as this is not actually extended anywhere outside the package. Dont feel very strongly, but should we leave it as it for now and change the visibility if its required to be  used outside the package?